packages/vega-util/src/extend.ts

```diff
@@ -0,0 +1,12 @@
+export type Writable = Record<string, unknown>;
+
+export default function extend<T extends Writable>(_: T, ...args: Writable[]): T {
+  for (let i = 0; i < args.length; ++i) {
+    const x = args[i];
+    for (const k in x) {
+      (_ as Writable)[k] = x[k];
```

Comment:
`as any` was also an option here since we don't actualy check if _ is an object, but this avoids a runtime behavior change so I figured start here

About the type, technically the return type is T + more properties, but we don't know in advance what properties the extended objects are going to add to the base.